### DISCUSSION TITLE
pointer intersection with bounds

### DISCUSSION THREAD

**Discussion by mludlum** [StereoKit/StereoKit#1087] [Q&A] [answered]
Trying to make a paddle game.  I want to limit the paddle to run along a track.  I know the track is in the right place because I rendered a Mesh there.  Idea is that I find intersection of track with pointer aim and put the paddle there.    I can kinda get this to work by using eye tracking in the simulator (which works when I shift-move-mouse), but I want this to work with a controller on the Quest 2/3.  I'm using your most recent preview build because I wanted to see the pointer line, but it looks like the cursor tracking is limited to UI? 

Can you tell me what I'm doing wrong in the step code?

```
// SETUP
Bounds ballTrack = Bounds.FromCorners(new Vec3(StageLeftTopFar.x, PaddleBottom, -PaddleDims.z / 2), new Vec3(StageRightBottomNear.x, PaddleBottom + Brick.BrickHeight, PaddleDims.z / 2));
            Mesh ballTrackMesh = Mesh.GenerateCube(ballTrack.dimensions);

Controller c = Input.Controller(Handed.Right);
Vec3 paddlePosition = ballTrack.center;


//STEP
if (ballTrack.Intersect(Input.Eyes.Ray, out Vec3 at)) // this works in simulator
{
    paddlePosition = new Vec3(at.x, ballTrack.center.y, ballTrack.center.z);
}
else if (ballTrack.Intersect(c.aim.Ray, out Vec3 at2))  //  this does not work in simulator nor on the Quest 3
{
    paddlePosition = new Vec3(at2.x, ballTrack.center.y, ballTrack.center.z);
}
paddlle.Draw(Material.Default, Matrix.T(paddlePosition), Color.Black);


```

**Comment by mludlum** [comment, marked as answer]
I had to put this in the Step loop to get this to work:

c = Input.Controller(Handed.Right);

**Reply by maluoi** [reply]
Ah hey! Glad you found it! SK does return a decent number of things like this by value as structs, so a good thing to keep an eye out for!
